packages/wallets/alpha/src/adapter.ts

```diff
@@ -106,6 +107,7 @@ export class AlphaWalletAdapter extends BaseMessageSignerWalletAdapter {
 
             // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
             const wallet = window.alpha!;
+            if (!wallet) throw new WalletNotReadyError();
```

Comment:
The reason these property checks aren't _strictly_ necessary (and why we use the `!` non-null assertion) is because we already checked for `window.alpha` in the polling mechanism of the constructor to update the `readyState` to `installed`. Since we already checked this in `connect` function, we are relying on it not to have changed, but I'm not aware of a case where this assumption is faulty.